src/Cache/SysVCacheItemPool.php

```diff
@@ -0,0 +1,223 @@
+
+    /** @var int */
+    private $sysvKey;
+
+    /**
+     * @var CacheItemInterface[]
+     */
+    private $items;
+
+    /**
+     * @var CacheItemInterface[]
+     */
+    private $deferredItems;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * Save the current items.
+     *
+     * @return bool true when success, false upon failure
+     */
+    private function saveCurrentItems()
+    {
+        $shmid = shm_attach(
+            $this->sysvKey,
+            $this->options['memsize'],
+            $this->options['perm']
+        );
+        if ($shmid !== false) {
+            $ret = shm_put_var(
+                $shmid,
+                $this->options['variableKey'],
+                $this->items
+            );
+            shm_detach($shmid);
+            return $ret;
+        }
+        return false;
+    }
+
+    /**
+     * Load the items from the shared memory.
+     *
+     * @return bool true when success, false upon failure
+     */
+    private function loadItems()
+    {
+        $shmid = shm_attach(
+            $this->sysvKey,
+            $this->options['memsize'],
+            $this->options['perm']
+        );
+        if ($shmid !== false) {
+            $data = @shm_get_var($shmid, $this->options['variableKey']);
+            if (!empty($data)) {
+                $this->items = $data;
+            } else {
+                $this->items = [];
+            }
+            shm_detach($shmid);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Create a SystemV shared memory based CacheItemPool.
+     *
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type int $variableKey The variable key for getting the data from
+     *           the shared memory. **Defaults to** 1.
+     *     @type string $proj The project identifier for ftok. This needs to
+     *           be a one character string. **Defaults to** 'A'.
+     *     @type int $memsize The memory size in bytes for shm_attach.
+     *           **Defaults to** 10000.
+     *     @type int $perm The permission for shm_attach. **Defaults to** 0600.
+     */
+    public function __construct($options = [])
+    {
+        if (! extension_loaded('sysvshm')) {
+            throw \RuntimeException(
+                'sysvshm extension is required to use this ItemPool');
+        }
+        $this->options = $options + [
+            'variableKey' => self::VAR_KEY,
+            'proj' => self::DEFAULT_PROJ,
+            'memsize' => self::DEFAULT_MEMSIZE,
+            'perm' => self::DEFAULT_PERM
+        ];
+        $this->items = [];
+        $this->deferredItems = [];
+        $this->sysvKey = ftok(__FILE__, $this->options['proj']);
+        $this->loadItems();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $this->loadItems();
+        return current($this->getItems([$key]));
+    }
+    /**
```

Comment:
nit but, let's consistently put a newline between the ending curly of one function and the start of the next one (several times below as well).